java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -788,6 +790,50 @@ Resolution unresolved() {
     return resolution;
   }
 
+  public JavaType conditionalExpressionType(JavaType trueType, JavaType falseType) {
+    if (trueType == falseType) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)